File: src/test/java/de/pflugradts/passbird/application/commandhandling/InputHandlerTestFactory.java
package de.pflugradts.passbird.application.commandhandling;

import de.pflugradts.passbird.application.commandhandling.command.CommandFactory;
import de.pflugradts.passbird.application.commandhandling.command.namespace.NamespaceCommandFactory;
import de.pflugradts.passbird.application.commandhandling.handler.CommandHandler;

import java.util.Set;

class InputHandlerTestFactory {

    static InputHandler setupInputHandlerFor(final CommandBus commandBus) {
        return new InputHandler(
            commandBus,
            new CommandFactory(new NamespaceCommandFactory()));
    }

    static InputHandler setupInputHandlerFor(final CommandHandler commandHandler) {
        return new InputHandler(
            new CommandBus(Set.of(commandHandler)),
            new CommandFactory(new NamespaceCommandFactory()));
    }

}

File: src/test/kotlin/de/pflugradts/passbird/application/commandhandling/InputHandlerTestFactory.kt
package de.pflugradts.passbird.application.commandhandling

import de.pflugradts.passbird.application.commandhandling.handler.CommandHandler

private val commandFactory = CommandFactory(
    memoryCommandFactory = MemoryCommandFactory(),
    nestCommandFactory = NestCommandFactory(),
    proteinCommandFactory = ProteinCommandFactory(),
    setCommandFactory = SetCommandFactory(),
)

fun createInputHandlerFor(commandBus: CommandBus) = InputHandler(commandBus, commandFactory)
fun createInputHandlerFor(commandHandler: CommandHandler) = InputHandler(CommandBus(setOf(commandHandler)), commandFactory)
